traits/traits_listener.py

```diff
@@ -1024,6 +1024,10 @@ def parse(self):
         # a shortcut too. Whitespace should already have been stripped from the
         # start and end.
 
+        if self.text.strip().endswith(","):
+            raise TraitNotificationError("Error parsing name. "
```

Comment:
Please use `self.error` instead in order to raise an exception type consistent with `parse_group`.